python/cudf/cudf/core/_base_index.py

```diff
@@ -1911,6 +1911,12 @@ def from_pandas(cls, index: pd.Index, nan_as_null=no_default):
                 step=index.step,
                 name=index.name,
             )
+        elif isinstance(index, pd.DatetimeIndex):
```

Comment:
So we were just going `as_column` hence having no visibility into `freq`? Can you check if we run into this issue when we do `cudf.Index(pd.DatetimeIndex(..))` too?

Reply:
Might aswell check `as_index` flow too while you are at it.